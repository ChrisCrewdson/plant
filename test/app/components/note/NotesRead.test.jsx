
const React = require('react');
const renderer = require('react-test-renderer');
const MuiThemeProvider = require('material-ui/styles/MuiThemeProvider').default;
const getMuiTheme = require('material-ui/styles/getMuiTheme').default;
const lightBaseTheme = require('material-ui/styles/baseThemes/lightBaseTheme').default;
const {
  MemoryRouter,
} = require('react-router-dom');
const { Provider } = require('react-redux');
const NotesRead = require('../../../../app/components/note/NotesRead');
const store = require('../../../../app/store');
const App = require('../../../../app/components/App');

const muiTheme = getMuiTheme(lightBaseTheme);

describe('NotesRead', () => {
  const storeDispatch = store.dispatch;

  beforeAll(() => {
    store.dispatch = () => {};
  });

  afterAll(() => {
    store.dispatch = storeDispatch;
  });

  test('should render a NotesRead', () => {
    const interim = {
    };
    const locationId = 'l-1';
    const notes = {
      'n-1': {
        _id: 'n-1',
        date: 1,
        note: 'This is note text',
        plantIds: ['p-1'],
        title: 'Note #1',
      },
    };
    const plant = {
      _id: 'p-1',
      notes: ['n-1'],
    };

    const plants = {
      'p-1': plant,
    };

    const component = renderer.create(
      <MuiThemeProvider muiTheme={muiTheme}>
        <Provider store={store}>
          <App>
            <MemoryRouter>
              <NotesRead
                dispatch={storeDispatch}
                interim={interim}
                locationId={locationId}
                notes={notes}
                plant={plant}
                plants={plants}
                userCanEdit
              />
            </MemoryRouter>
          </App>
        </Provider>
      </MuiThemeProvider>);
    const badgeTotals = component.toJSON();
    expect(badgeTotals).toMatchSnapshot();
  });
});
