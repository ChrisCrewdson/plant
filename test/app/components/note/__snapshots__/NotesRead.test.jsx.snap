// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotesRead should render a NotesRead 1`] = `
<div
  className="react-root"
>
  <div>
    <div
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#ffffff",
          "borderRadius": 2,
          "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "margin": 20,
          "muiPrepared": true,
          "padding": 20,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "width": "100%",
        }
      }
    >
      <div
        id="n-1"
      >
        <a
          href="?noteid=n-1#n-1"
        >
          <svg
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "display": "inline-block",
                "fill": "currentColor",
                "height": 24,
                "muiPrepared": true,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "width": 24,
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5z"
            />
          </svg>
        </a>
      </div>
      <h5>
        01-Dec-1899 (119 years ago)
      </h5>
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>This is note text</p>
",
          }
        }
      />
      <h2
        className="vcenter"
      >
        <div
          style={
            Object {
              "textAlign": "right",
            }
          }
        >
          <div
            className={undefined}
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                "backgroundColor": "transparent",
                "borderRadius": "50%",
                "boxShadow": "0 3px 10px rgba(0, 0, 0, 0.16),
         0 3px 10px rgba(0, 0, 0, 0.23)",
                "boxSizing": "border-box",
                "color": "rgba(0, 0, 0, 0.87)",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "muiPrepared": true,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              }
            }
          >
            <button
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              style={
                Object {
                  "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                  "backgroundColor": "#00bcd4",
                  "border": 10,
                  "borderRadius": "50%",
                  "boxSizing": "border-box",
                  "cursor": "pointer",
                  "display": "inline-block",
                  "fontFamily": "Roboto, sans-serif",
                  "fontSize": "inherit",
                  "fontWeight": "inherit",
                  "height": 56,
                  "margin": 0,
                  "muiPrepared": true,
                  "outline": "none",
                  "overflow": "hidden",
                  "padding": 0,
                  "position": "relative",
                  "textAlign": "center",
                  "textDecoration": "none",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "verticalAlign": "bottom",
                  "width": 56,
                }
              }
              tabIndex={0}
              title="Edit"
              type="button"
            >
              <div
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
              >
                <div
                  style={
                    Object {
                      "muiPrepared": true,
                      "top": 0,
                      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    }
                  }
                >
                  <svg
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    style={
                      Object {
                        "color": "#ffffff",
                        "display": "inline-block",
                        "fill": "#ffffff",
                        "height": 56,
                        "lineHeight": "56px",
                        "muiPrepared": true,
                        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                        "userSelect": "none",
                        "width": 24,
                      }
                    }
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                    />
                  </svg>
                </div>
              </div>
            </button>
          </div>
          <div
            className={undefined}
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                "backgroundColor": "transparent",
                "borderRadius": "50%",
                "boxShadow": "0 3px 10px rgba(0, 0, 0, 0.16),
         0 3px 10px rgba(0, 0, 0, 0.23)",
                "boxSizing": "border-box",
                "color": "rgba(0, 0, 0, 0.87)",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "marginLeft": "10px",
                "muiPrepared": true,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              }
            }
          >
            <button
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              style={
                Object {
                  "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                  "backgroundColor": "#ff4081",
                  "border": 10,
                  "borderRadius": "50%",
                  "boxSizing": "border-box",
                  "cursor": "pointer",
                  "display": "inline-block",
                  "fontFamily": "Roboto, sans-serif",
                  "fontSize": "inherit",
                  "fontWeight": "inherit",
                  "height": 56,
                  "margin": 0,
                  "muiPrepared": true,
                  "outline": "none",
                  "overflow": "hidden",
                  "padding": 0,
                  "position": "relative",
                  "textAlign": "center",
                  "textDecoration": "none",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "verticalAlign": "bottom",
                  "width": 56,
                }
              }
              tabIndex={0}
              title="Delete"
              type="button"
            >
              <div
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
              >
                <div
                  style={
                    Object {
                      "muiPrepared": true,
                      "top": 0,
                      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    }
                  }
                >
                  <svg
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    style={
                      Object {
                        "color": "#ffffff",
                        "display": "inline-block",
                        "fill": "#ffffff",
                        "height": 56,
                        "lineHeight": "56px",
                        "muiPrepared": true,
                        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                        "userSelect": "none",
                        "width": 24,
                      }
                    }
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                    />
                  </svg>
                </div>
              </div>
            </button>
          </div>
        </div>
      </h2>
    </div>
  </div>
</div>
`;
